### ISSUE TITLE
StopClient() is causing crash

### ISSUE THREAD

**Issue by Lymdun** [MirrorNG/MirrorNG#68]
**Describe the bug**
When calling **StopClient()**, the entire game is crashing. Also the client isn't properly disconnected from the server.

**Expected behavior**
Client should be properly disconnected from the server, and no crash should be involved.

**Comment by Lymdun** [comment]
Stacktrace:

```
StackOverflowException: The requested operation caused a stack overflow.
UnityEngine.Object.IsNativeObjectAlive (UnityEngine.Object o) (at C:/buildslave/unity/build/Runtime/Export/Scripting/UnityEngineObject.bindings.cs:156)
UnityEngine.Object.CompareBaseObjects (UnityEngine.Object lhs, UnityEngine.Object rhs) (at C:/buildslave/unity/build/Runtime/Export/Scripting/UnityEngineObject.bindings.cs:135)
UnityEngine.Object.op_Inequality (UnityEngine.Object x, UnityEngine.Object y) (at C:/buildslave/unity/build/Runtime/Export/Scripting/UnityEngineObject.bindings.cs:396)
UnityEngine.Events.BaseInvokableCall.AllowInvoke (System.Delegate delegate) (at C:/buildslave/unity/build/Runtime/Export/UnityEvent/UnityEvent.cs:133)
UnityEngine.Events.InvokableCall.Invoke () (at C:/buildslave/unity/build/Runtime/Export/UnityEvent/UnityEvent.cs:165)
UnityEngine.Events.UnityEvent.Invoke () (at C:/buildslave/unity/build/Runtime/Export/UnityEvent/UnityEvent/UnityEvent_0.cs:58)
Mirror.Tcp.TcpTransport.<Awake>b__5_4 () (at Assets/Mirror/Runtime/Transport/Tcp/TcpTransport.cs:33)
Mirror.Tcp.Client.Disconnect () (at Assets/Mirror/Runtime/Transport/Tcp/Client.cs:89)
Mirror.Tcp.TcpTransport.ClientDisconnect () (at Assets/Mirror/Runtime/Transport/Tcp/TcpTransport.cs:54)
Mirror.NetworkConnectionToServer.Disconnect () (at Assets/Mirror/Runtime/NetworkConnectionToServer.cs:29)
Mirror.NetworkClient.Disconnect () (at Assets/Mirror/Runtime/NetworkClient.cs:196)
Mirror.NetworkManager.StopClient () (at Assets/Mirror/Runtime/NetworkManager.cs:603)
Mirror.NetworkManager.OnClientDisconnect (Mirror.NetworkConnection conn) (at Assets/Mirror/Runtime/NetworkManager.cs:1359)
Mirror.NetworkManager.OnClientDisconnectInternal (Mirror.NetworkConnection conn, Mirror.DisconnectMessage msg) (at Assets/Mirror/Runtime/NetworkManager.cs:1184)
Mirror.MessagePacker+<>c__DisplayClass6_0`2[T,C].<MessageHandler>b__0 (Mirror.NetworkConnection conn, Mirror.NetworkReader reader, System.Int32 channelId) (at Assets/Mirror/Runtime/MessagePacker.cs:133)
Mirror.NetworkConnection.InvokeHandler (System.Int32 msgType, Mirror.NetworkReader reader, System.Int32 channelId) (at Assets/Mirror/Runtime/NetworkConnection.cs:215)
Mirror.NetworkConnection.InvokeHandler[T] (T msg, System.Int32 channelId) (at Assets/Mirror/Runtime/NetworkConnection.cs:243)
Mirror.NetworkClient.OnDisconnected () (at Assets/Mirror/Runtime/NetworkClient.cs:150)
UnityEngine.Events.InvokableCall.Invoke () (at C:/buildslave/unity/build/Runtime/Export/UnityEvent/UnityEvent.cs:166)
UnityEngine.Events.UnityEvent.Invoke () (at C:/buildslave/unity/build/Runtime/Export/UnityEvent/UnityEvent/UnityEvent_0.cs:58)
Mirror.Tcp.TcpTransport.<Awake>b__5_4 () (at Assets/Mirror/Runtime/Transport/Tcp/TcpTransport.cs:33)
Mirror.Tcp.Client.Disconnect () (at Assets/Mirror/Runtime/Transport/Tcp/Client.cs:89)
Mirror.Tcp.TcpTransport.ClientDisconnect () (at Assets/Mirror/Runtime/Transport/Tcp/TcpTransport.cs:54)
Mirror.NetworkConnectionToServer.Disconnect () (at Assets/Mirror/Runtime/NetworkConnectionToServer.cs:29)
Mirror.NetworkClient.Disconnect () (at Assets/Mirror/Runtime/NetworkClient.cs:196)
Mirror.NetworkManager.StopClient () (at Assets/Mirror/Runtime/NetworkManager.cs:603)
Mirror.NetworkManager.OnClientDisconnect (Mirror.NetworkConnection conn) (at Assets/Mirror/Runtime/NetworkManager.cs:1359)
Mirror.NetworkManager.OnClientDisconnectInternal (Mirror.NetworkConnection conn, Mirror.DisconnectMessage msg) (at Assets/Mirror/Runtime/NetworkManager.cs:1184)
Mirror.MessagePacker+<>c__DisplayClass6_0`2[T,C].<MessageHandler>b__0 (Mirror.NetworkConnection conn, Mirror.NetworkReader reader, System.Int32 channelId) (at Assets/Mirror/Runtime/MessagePacker.cs:133)
Mirror.NetworkConnection.InvokeHandler (System.Int32 msgType, Mirror.NetworkReader reader, System.Int32 channelId) (at Assets/Mirror/Runtime/NetworkConnection.cs:215)
Mirror.NetworkConnection.InvokeHandler[T] (T msg, System.Int32 channelId) (at Assets/Mirror/Runtime/NetworkConnection.cs:243)
Mirror.NetworkClient.OnDisconnected () (at Assets/Mirror/Runtime/NetworkClient.cs:150)
UnityEngine.Events.InvokableCall.Invoke () (at C:/buildslave/unity/build/Runtime/Export/UnityEvent/UnityEvent.cs:166)
UnityEngine.Events.UnityEvent.Invoke () (at C:/buildslave/unity/build/Runtime/Export/UnityEvent/UnityEvent/UnityEvent_0.cs:58)
...
```

**Comment by paulpach** [comment]
fixed in f584388a16e746ac5c3000123a02a5c77387765e
